crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -224,33 +226,59 @@ impl SettlementEncoder {
         // <=> order.sellAmount.mul(sellPrice)  >= order.buyAmount.mul(buyPrice)
         // <=> equation from smart contract
 
-        let buy_price = self
-            .clearing_prices
+        self.clearing_prices
             .get(&order.data.sell_token)
-            .context("settlement missing sell token")?
+            .unwrap_or(&order.data.buy_amount)
             .checked_mul(order.data.sell_amount)
             .context("buy_price calculation failed")?
             .checked_div(order.data.buy_amount)
-            .context("buy_price calculation failed")?;
+            .context("buy_price calculation failed")
+    }
+
+    fn add_custom_price_trade(
+        &mut self,
+        order: Order,
+        executed_amount: U256,
+        scaled_unsubsidized_fee: U256,
+        buy_price: U256,
+    ) -> Result<TradeExecution> {
+        verify_executed_amount(&order, executed_amount)?;
+        // For the encoding strategy of liquidity and limit orders, the sell prices are taken from
+        // the uniform clearing price vector. Therefore, either there needs to be an existing price
+        // for the sell token in the uniform clearing prices or we have to create a new price entry beforehand,
+        // if the sell token price is not yet available
+        if self.token_index(order.data.sell_token) == None {
```

Comment:
I'd prefer `.is_none()`. But this isn't important. Wish clippy linted this.

Reply:
It does in 1.65.0, so we should use `is_none()` lest need a separate PR to fix the lint.

Reply:
Added in [8d5ccf3](https://github.com/cowprotocol/services/pull/712/commits/8d5ccf3d31a9b1d7755492f4d5e76a279d9127dd).